PhysicsTools/NanoAOD/python/taus_cff.py

```diff
@@ -66,8 +68,8 @@ def _tauIdWPMask(pattern, choices, doc="", from_raw=False, wp_thrs=None):
        leadTkDeltaEta = Var("?leadChargedHadrCand.isNonnull()?(leadChargedHadrCand.eta - eta):0",float, doc="eta of the leading track, minus tau eta",precision=8),
        leadTkDeltaPhi = Var("?leadChargedHadrCand.isNonnull()?deltaPhi(leadChargedHadrCand.phi, phi):0",float, doc="phi of the leading track, minus tau phi",precision=8),
 
-       dxy = Var("?leadChargedHadrCand.isNonnull()?leadChargedHadrCand().dxy():0",float, doc="d_{xy} of lead track with respect to PV, in cm (with sign)",precision=10),
-       dz = Var("?leadChargedHadrCand.isNonnull()?leadChargedHadrCand().dz():0",float, doc="d_{z} of lead track with respect to PV, in cm (with sign)",precision=14),
+       dxy = Var("?leadChargedHadrCand.isNonnull()?leadChargedHadrCand().dxy():0",float, doc="d_{xy} of lead track with respect to PV, in cm (with sign)",precision=10, lazyEval=True),
```

Comment:
Why are `dxy` and `dz` marked `lazyEval` but ` leadTkPtOverTauPt`, `leadTkDeltaEta` and ` leadTkDeltaPhi` which use the same variables not marked?

Reply:
@hqucms ping

Reply:
It's because leadChargedHadrCand() returns the base type `reco::CandidatePtr` and it needs to be casted to `PackedCandidate` to call dxy() / dz(). For simple pt/eta it's not needed. 
I added a comment in the code.